onnxruntime/core/util/math_cpu.cc

```diff
@@ -201,6 +201,14 @@ void GemmEx<float, ThreadPool>(CBLAS_TRANSPOSE TransA, CBLAS_TRANSPOSE TransB, p
   MlasGemm(TransA, TransB, M, N, K, alpha, A, lda, B, ldb, beta, C, ldc, threadpool);
 }
 
+// bugbug
+template <>
+void GemmEx<MLFloat16, ThreadPool>(CBLAS_TRANSPOSE, CBLAS_TRANSPOSE, ptrdiff_t, ptrdiff_t, ptrdiff_t,
+                                   MLFloat16, const MLFloat16*, int, const MLFloat16*, int, MLFloat16,
+                                   MLFloat16*, int, ThreadPool*) {
```

Comment:
Could we support the following:
A FP16, B FP16, C FP32  -- for QxK
A FP32, B FP16, C FP16  -- for SxV

Reply:
We don't have fp16 matmul implemented on CPU. I am thinking it may be better to convert the input and kv cache to fp16 in GQAAttention